### PR TITLE
Fix(chain): fix apply chunks

### PR DESCRIPTION
**Pull request by bowenwang1996**
Port https://github.com/near/nearcore/pull/4226 to master.

Test plan
----------
Nayduck http://nayduck.eastus.cloudapp.azure.com:3000/#/run/1508

### CODE REVIEW

File: runtime/runtime/src/lib.rs [hunk 1]
```diff
@@ -1098,6 +1100,23 @@ impl Runtime {
                 &mut stats,
             )?;
         }
+        if !apply_state.is_new_chunk
+            && apply_state.current_protocol_version
+                >= ProtocolFeature::FixApplyChunks.protocol_version()
```
Comment (matklad) [inline runtime/runtime/src/lib.rs:1105]: This is unrelated to the PR, but we could (and maybe should) fix the API discrepancy between stabe & unstable features like this:

```rust
diff --git a/core/primitives/Cargo.toml b/core/primitives/Cargo.toml
index 92d31a613..438301be6 100644
--- a/core/primitives/Cargo.toml
+++ b/core/primitives/Cargo.toml
@@ -39,10 +39,8 @@ near-vm-errors = { path = "../../runtime/near-vm-errors" }
 near-rpc-error-macro = { path = "../../tools/rpctypegen/macro" }
 
 [features]
-default = ["jemallocator", "stable_protocol_features"]
+default = ["jemallocator"]
 dump_errors_schema = ["near-rpc-error-macro/dump_errors_schema"]
-stable_protocol_features = ["protocol_feature_lower_storage_cost"]
-protocol_feature_lower_storage_cost = []
 protocol_feature_add_account_versions = ["near-primitives-core/protocol_feature_add_account_versions"]
 protocol_feature_forward_chunk_parts = []
 protocol_feature_rectify_inflation = []
diff --git a/core/primitives/src/runtime/config.rs b/core/primitives/src/runtime/config.rs
index c3494725a..be67b0b32 100644
--- a/core/primitives/src/runtime/config.rs
+++ b/core/primitives/src/runtime/config.rs
@@ -60,7 +60,7 @@ impl RuntimeConfig {
         protocol_version: ProtocolVersion,
     ) -> Arc<Self> {
         if checked_feature!(
-            "protocol_feature_lower_storage_cost",
+            "stable",
             LowerStorageCost,
             protocol_version
         ) {
diff --git a/core/primitives/src/version.rs b/core/primitives/src/version.rs
index 8b2282e3c..c28dec1db 100644
--- a/core/primitives/src/version.rs
+++ b/core/primitives/src/version.rs
@@ -85,7 +85,6 @@ pub enum ProtocolFeature {
     #[cfg(feature = "protocol_feature_block_header_v3")]
     BlockHeaderV3,
     /// Decreases the storage cost of 1 byte by 10X.
-    #[cfg(feature = "protocol_feature_lower_storage_cost")]
     LowerStorageCost,
     #[cfg(feature = "protocol_feature_alt_bn128")]
     AltBn128,
@@ -111,7 +110,6 @@ impl ProtocolFeature {
     pub const fn protocol_version(self) -> ProtocolVersion {
         match self {
             // Stable features
-            #[cfg(feature = "protocol_feature_lower_storage_cost")]
             ProtocolFeature::LowerStorageCost => 42,
             ProtocolFeature::DeleteActionRestriction => 43,
             ProtocolFeature::FixApplyChunks => 44,
@@ -139,6 +137,10 @@ impl ProtocolFeature {
 
 #[macro_export]
 macro_rules! checked_feature {
+    ("stable", $feature:ident, $current_protocol_version:expr) => {{
+        $crate::version::ProtocolFeature::$feature.protocol_version() <= $current_protocol_version
+    }};
+
     ($feature_name:tt, $feature:ident, $current_protocol_version:expr) => {{
         #[cfg(feature = $feature_name)]
         let is_feature_enabled = $crate::version::ProtocolFeature::$feature.protocol_version()
diff --git a/runtime/runtime/src/lib.rs b/runtime/runtime/src/lib.rs
index 092dad9b2..68c5206d1 100644
--- a/runtime/runtime/src/lib.rs
+++ b/runtime/runtime/src/lib.rs
@@ -7,6 +7,7 @@ use log::debug;
 pub use near_crypto;
 use near_crypto::PublicKey;
 pub use near_primitives;
+use near_primitives::checked_feature;
 use near_primitives::runtime::get_insufficient_storage_stake;
 use near_primitives::{
     account::{AccessKey, Account},
@@ -1101,8 +1102,7 @@ impl Runtime {
             )?;
         }
         if !apply_state.is_new_chunk
-            && apply_state.current_protocol_version
-                >= ProtocolFeature::FixApplyChunks.protocol_version()
+            && checked_feature!("stable", FixApplyChunks, apply_state.current_protocol_version)
         {
             let (trie_changes, state_changes) = state_update.finalize()?;
             let proof = trie.recorded_storage();
```

I'll look into that after  this PR is merged

Comment (bowenwang1996) [reply runtime/runtime/src/lib.rs:1105]: @matklad Thanks!

**Review by SkidanovAlex — APPROVED** [review]

**Review by frol — APPROVED** [review]

**Comment by khorolets** [comment]
@bowenwang1996 FYI we had to tweak `chain.rs` to got execution outcomes on indexer side like this. Please, consider to include it in this PR or let me know if you want me to add it within my updates.

```diff
diff --git a/chain/chain/src/chain.rs b/chain/chain/src/chain.rs
index 194e4a7b..9d3e0081 100644
--- a/chain/chain/src/chain.rs
+++ b/chain/chain/src/chain.rs
@@ -2031,8 +2031,8 @@ impl Chain {
         let chunk_headers = block
             .chunks()
             .iter()
-            .filter_map(
-                |h| if h.height_included() == block_height { Some(h.clone()) } else { None },
+            .map(
+                |h| h.clone() // if h.height_included() == block_height { Some(h.clone()) } else { None },
             )
             .collect::<Vec<_>>();
```

**Review by EgorKulikov — APPROVED** [review]

**Comment by bowenwang1996** [comment]
> @bowenwang1996 FYI we had to tweak chain.rs to got execution outcomes on indexer side like this. Please, consider to include it in this PR or let me know if you want me to add it within my updates.

I thought you are going to do this in your PR, so didn't include it here

File: neard/res/genesis_config.json [hunk 2]
```diff
@@ -1,239 +1,239 @@
...
-            "send_sir": 1925331, 
-            "send_not_sir": 1925331, 
+            "send_sir": 1925331,
+            "send_not_sir": 1925331,
             "execution": 1925331
           }
-        }, 
+        },
         "delete_key_cost": {
-          "send_sir": 94946625000,
```
Comment (olonho) [inline neard/res/genesis_config.json:111]: Is change in costs pure trailing space removal?

Comment (bowenwang1996) [reply neard/res/genesis_config.json:111]: Yeah it is auto-generated by the script

**Review by olonho — APPROVED** [review]

**Pull request merged**